tests/AssetBundleTest.php

```diff
@@ -0,0 +1,385 @@
+            ],
+            $this->assetManager->getJsFiles()[$urlJs]['attributes']
+        );
+    }
+
+    public function testBasePath(): void
+    {
+        $this->assertEmpty($this->assetManager->getAssetBundles());
+
+        $this->assetManager->register([BaseAsset::class]);
+
+        $urlCss = '/baseUrl/css/basePath.css';
+        $urlJs = '/baseUrl/js/basePath.js';
+
+        $this->assertStringContainsString(
+            $urlCss,
+            $this->assetManager->getCssFiles()[$urlCss]['url']
+        );
+        $this->assertEquals(
+            [
+                'integrity' => 'integrity-hash',
+                'crossorigin' => 'anonymous'
+            ],
+            $this->assetManager->getCssFiles()[$urlCss]['attributes']
+        );
+
+        $this->assertStringContainsString(
+            $urlJs,
+            $this->assetManager->getJsFiles()[$urlJs]['url']
+        );
+        $this->assertEquals(
+            [
+                'integrity' => 'integrity-hash',
+                'crossorigin' => 'anonymous',
+                'position' => 3
+            ],
+            $this->assetManager->getJsFiles()[$urlJs]['attributes']
+        );
+    }
+
+    public function testBasePathEmptyException(): void
+    {
+        $this->assetManager->setBundles(
+            [
+                BaseAsset::class => [
+                    'basePath' => null,
+                ]
+            ]
+        );
+
+        $this->assertEmpty($this->assetManager->getAssetBundles());
+
+        $message = 'basePath must be set in AssetManager->setBasePath($path) or ' .
+            'AssetBundle property public ?string $basePath = $path';
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage($message);
+
+        $this->assetManager->register([BaseAsset::class]);
+    }
+
+    public function testBaseUrlEmptyException(): void
+    {
+        $this->assetManager->setBundles(
+            [
+                BaseAsset::class => [
+                    'basePath' => null,
+                    'baseUrl' => null,
+                ]
+            ]
+        );
+
+        $this->assetManager->setBasePath('@basePath');
+
+        $this->assertEmpty($this->assetManager->getAssetBundles());
+
+        $message = 'baseUrl must be set in AssetManager->setBaseUrl($path) or ' .
+            'AssetBundle property public ?string $baseUrl = $path';
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage($message);
+
+        $this->assetManager->register([BaseAsset::class]);
+    }
+
+    public function testBasePathEmptyWithAssetManagerSetBasePath(): void
+    {
+        $config = [
+            'basePath' => null,
+        ];
+
+        $this->assetManager->setBasePath('@basePath');
+
+        $this->assertEmpty($this->assetManager->getAssetBundles());
+        $this->assertIsObject($this->assetManager->getBundle(BaseAsset::class, $config, false));
```

Comment:
There's only one argument in current `getBundle()`.

Reply:
Done.